internal/generator/generator_util.go

```diff
@@ -184,12 +184,39 @@ func getMasksFromPath(path string) (directive2Masks map[string][]mask, err error
 	return directive2Masks, nil
 }
 
-func genFromSrcCode(codePath string, mapVariableName string, matchFnName string, writer io.Writer) error {
+func genFromSrcCode(codePath string, mapVariableName string, matchFnName string, writer io.Writer,
+	filter map[string]struct{}, override map[string][]mask) error {
 	directive2Masks, err := getMasksFromPath(codePath)
 	if err != nil {
 		return err
 	}
 
+	if filter != nil {
+		dropDirectives := []string{}
```

Comment:
```suggestion
		var dropDirectives []string
```

Avoids allocation in the case that there are no values in `directive2Masks`. Remember that you can `append()` a `nil` slice as it will allocate as needed and return the new slice.

Reply:
fixed